### PR TITLE
Use tox to manage test environment

### PR DESCRIPTION
**Pull request by NouemanKHAL**

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    click==7.1.2
+    click-config-file==0.6.0
+    requests==2.25.1
+    deepdiff==5.5.0
+    black>=20.8b1
+    pytest>=5.2.4
+    pytest-black
+    pytest-console-scripts
+    pytest-recording
+    pytest-env
```
Comment (jirikuncar) [inline tox.ini:24]:
```suggestion
extras = tests
```

Comment (jirikuncar) [inline tox.ini:24]: can you update extra dependencies in `setup.cfg`?

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,25 @@
+[tox]
+minversion = 3.7.0
+skip_missing_interpreters = true
+envlist =
+    py{37,38,39}
+
+[testenv]
+passenv = DD_SOURCE_*  DD_DESTINATION_*
```
Comment (jirikuncar) [inline tox.ini:8]:
```suggestion
passenv = DD_SOURCE_*  DD_DESTINATION_* PYTEST_*
```
to allow for `PYTEST_ADOPTS="--ddtrace"`

File: tox.ini (hunk 1 above)
Comment (NouemanKHAL) [reply tox.ini]: nothing to update there apparently

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+[testenv]
+passenv = DD_SOURCE_*  DD_DESTINATION_* PYTEST_*
+setenv =
+    DD_HTTP_CLIENT_RETRY_TIMEOUT = 300
+    MAX_WORKERS = 1
+    RECORD = {env:RECORD:false}
+usedevelop = true
+extras = tests
+commands = pytest -ra -v
```
Comment (zippolyte) [inline tox.ini:15]:
```suggestion
commands = pytest -ra -v {posargs}
```
to be able to specify arbitrary pytest args when invoking tox

**Review by zippolyte — APPROVED** [review]

**Pull request merged**